cli/src/stake.rs

```diff
@@ -2474,7 +2472,12 @@ pub fn get_epoch_boundary_timestamps(
                 break block_time;
             }
             Err(_) => {
-                epoch_start_slot += 1;
+                // TODO This is wrong.  We should not just increase the slot index if the RPC
+                // request failed.  It could have failed for a number of reasons, including, for
+                // example a network failure.
+                epoch_start_slot = epoch_start_slot
+                    .checked_add(1)
+                    .map_err(|_| "Reached last slot that fits into u64")?;
```

Comment:
This doesn't build because checked math returns `Option`s, so this should probably be something like:
```suggestion
                epoch_start_slot = epoch_start_slot
                    .checked_add(1)
                    .ok_or("Reached last slot that fits into u64")?;
```

Reply:
Thank you for catching it.  I should have compiled it, after I've replaced the saturating math %)